src/test/resources/com/puppycrawl/tools/checkstyle/grammar/InputRegressionJavaClass2.java

```diff
@@ -39,3 +39,5 @@ public InputRegressionJavaClass2 m2() {
     c5() { new c4().super(); }
     c5(int a) { new c4().<String>super(); }
 }
+interface i1 {}
+class c6<T extends c3<T> & i1> {}
```

Comment:
@pbludov You said `&` is like a list.
Would this list be `T extends c3<T>` and `i1`? Or is it `c3<T>`  and `i1` and both are children of `T`?

Reply:
I would rather avoid change in AST structure, we can discuss it in separate issue.

Reply:
@romani This whole issue is about changing the AST structure. This item is just confirming what is list and what is child/parent relationship.

Reply:
@pbludov ping